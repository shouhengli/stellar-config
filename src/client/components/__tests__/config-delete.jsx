const React = require('react');
const renderer = require('react-test-renderer');
const ConfigDelete = require('../config-delete.jsx');

describe('component config-delete', () => {
  test('enables yes button when confirmed', () => {
    const props = {
      confirmed: true,
      configType: 'graphSchema',
      configName: 'default',
      configDeleteName: 'default',
      handleNameChange: () => {},
      handleCloseButtonClick: () => {},
      handleDeleteButtonClick: () => {}
    };

    const component = renderer.create(<ConfigDelete {...props} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('disables yes button when not confirmed', () => {
    const props = {
      confirmed: false,
      configType: 'graphSchema',
      configName: 'default',
      configDeleteName: 'def',
      handleNameChange: () => {},
      handleCloseButtonClick: () => {},
      handleDeleteButtonClick: () => {}
    };

    const component = renderer.create(<ConfigDelete {...props} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
