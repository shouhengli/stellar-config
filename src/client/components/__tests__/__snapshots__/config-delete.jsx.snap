// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component config-delete disables yes button when not confirmed 1`] = `
<div
  className="message is-danger"
>
  <div
    className="message-header"
  >
    Confirmation
    <button
      className="delete"
      onClick={[Function]}
    />
  </div>
  <div
    className="message-body"
  >
    <p>
      Please type the configuration name to confirm the deletion.
    </p>
    <div
      className="field has-addons"
    >
      <div
        className="control is-expanded"
      >
        <input
          className="input is-danger"
          onChange={[Function]}
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="control"
      >
        <button
          className="button is-danger"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`component config-delete enables yes button when confirmed 1`] = `
<div
  className="message is-danger"
>
  <div
    className="message-header"
  >
    Confirmation
    <button
      className="delete"
      onClick={[Function]}
    />
  </div>
  <div
    className="message-body"
  >
    <p>
      Please type the configuration name to confirm the deletion.
    </p>
    <div
      className="field has-addons"
    >
      <div
        className="control is-expanded"
      >
        <input
          className="input is-danger"
          onChange={[Function]}
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="control"
      >
        <button
          className="button is-danger"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
